server/modules/environment-state/getInstanceState.js

```diff
@@ -90,7 +90,36 @@ module.exports = function getInstanceState(accountName, environmentName, nodeNam
       };
     }));
     // If undefined, it's not EM deployed service        
-    services = _.filter(services, service => service.DeploymentId !== undefined);
+    services = _.filter(services, (service) => service.DeploymentId !== undefined);
+
+    // Now make a diff with target state of Instance services
+    let targetServiceStates = yield serviceTargets.getAllServiceTargets(environmentName, runtimeServerRoleName);
+
+    // Primo, find any services that are in target state, but not on instance
+    _.each(targetServiceStates, (targetService) => {
+      if (_.find(services, { Name: targetService.Name }) === undefined && targetService.Action === Enums.ServiceAction.INSTALL) {
+        let missingService = {
+          Name: targetService.Name,
+          Version: targetService.Version,
+          Slice: targetService.Slice,
+          DeploymentId: targetService.DeploymentId,
+          Action: targetService.Action,
+          HealthChecks: [],
+          DiffWithTargetState: 'Missing',
+          Issues: { Warnings: [], Errors: [] }
+        };
+        missingService.Issues.Warnings.push(`Service that is in target state is missing`);
+        services.push(missingService);
+      }
+    });
+
+    // Secondo, find any services that are present on instance, but not in target state
+    _.each(services, (instanceService) => {
```

Comment:
Do we need to handle services that are present in the KV store but marked not to be installed?